consensus/abci_mempool_connection.go

```diff
@@ -0,0 +1,114 @@
+package consensus
```

Comment:
I feel your separation on `abci` connection went too far and it will be hard to find where to look at when we need to do changes on it.  I am not convinced this separation brings any actual value and I am sure it brings confusion. We should reconsider in putting all logic back together

Reply:
I answered that above. Damn github doesn't let me link a not review-published comment.